### PR TITLE
fix: change to dynamic PATH

### LINKED ISSUE #1346: refactor: `DefaultPathEnvUnix` should be dynamic
## Description

Currently, the `DefaultPathEnvUnix` is hard-coded with conda path and Julia path. Since we support different languages, this should be a dynamic one. It should not include the Julia path for Python environments.

---

**Message from the maintainers**:

Love this enhancement proposal? Give it a 👍. We prioritise the proposals with the most 👍.

### PR DESCRIPTION
**Pull request by kemingy**
* fix #1346 

## Usage

* runtime env
* build time path
  * init
  * renew when switch the user

## Design

* maintain it in the Graph (RuntimeEnvPaths)
* init as the system PATH
* update if needed

---

*Note*: runtime env PATH leaks to build time

### CODE REVIEW

File: pkg/lang/ir/v0/system.go [hunk 1]
```diff
@@ -54,7 +54,7 @@ func (g generalGraph) compileRun(root llb.State) llb.State {
 	}
 
 	workingDir := g.getWorkingDir()
-	stage := root.AddEnv("PATH", types.DefaultPathEnvUnix)
+	stage := root
```
Comment (gaocegege) [inline pkg/lang/ir/v0/system.go:57]: There is no need to keep stage.

File: pkg/lang/ir/v1/julia.go [hunk 2]
```diff
@@ -21,9 +21,11 @@ import (
 	"github.com/cockroachdb/errors"
 	"github.com/moby/buildkit/client/llb"
 	"github.com/sirupsen/logrus"
+	"github.com/tensorchord/envd/pkg/types"
 )
 
 func (g generalGraph) installJulia(root llb.State) (llb.State, error) {
```
Comment (gaocegege) [inline pkg/lang/ir/v1/julia.go:28]:
```suggestion
func (g *generalGraph) installJulia(root llb.State) (llb.State, error) {
```

File: pkg/lang/ir/v0/system.go (hunk 1 above)
Comment (kemingy) [reply pkg/lang/ir/v0/system.go]: Fixed

**Review by gaocegege — APPROVED** [review]
LGTM :+1:
/lgtm

**Pull request merged**